### PR TITLE
Mutually exclusive optional `HibernationSchedule.Start/End`

### PR DESCRIPTION
**Pull request by adracus**
**What this PR does / why we need it**:
Make `HibernationSchedule.Start/End` mutually exclusive. This allows to define HibernationSchedules that only wake up or only hibernate a Shoot (@petersutter thx for the idea).

**Special notes for your reviewer**:

**Release note**:

```improvement user
Either the `start` or `end` field in a shoot's hibernation schedule can now be omitted (although at least one of both needs to be specified). This allows to have schedules that only hibernate or only wake up a shoot.
```

### CODE REVIEW

File: pkg/apis/garden/v1beta1/types.go [hunk 1]
```diff
@@ -1192,11 +1192,12 @@ type Hibernation struct {
 
 // HibernationSchedule determines the hibernation schedule of a Shoot.
 // A Shoot will be regularly hibernated at each start time and will be woken up at each end time.
+// Start or End can be omitted, though at least one of each has to be specified.
 type HibernationSchedule struct {
 	// Start is a Cron spec at which time a Shoot will be hibernated.
-	Start string `json:"start"`
+	Start *string `json:"start"`
```
Comment (rfranzke) [inline pkg/apis/garden/v1beta1/types.go]: omitempty

File: pkg/apis/garden/v1beta1/types.go [hunk 2]
```diff
@@ -1192,11 +1192,12 @@ type Hibernation struct {
 
 // HibernationSchedule determines the hibernation schedule of a Shoot.
 // A Shoot will be regularly hibernated at each start time and will be woken up at each end time.
+// Start or End can be omitted, though at least one of each has to be specified.
 type HibernationSchedule struct {
 	// Start is a Cron spec at which time a Shoot will be hibernated.
```
Comment (rfranzke) [inline pkg/apis/garden/v1beta1/types.go:1197]: //+optional missing

File: pkg/apis/garden/types.go [hunk 3]
```diff
@@ -1200,11 +1200,12 @@ type Hibernation struct {
 
 // HibernationSchedule determines the hibernation schedule of a Shoot.
 // A Shoot will be regularly hibernated at each start time and will be woken up at each end time.
+// Start or End can be omitted, though at least one of each has to be specified.
 type HibernationSchedule struct {
 	// Start is a Cron spec at which time a Shoot will be hibernated.
```
Comment (rfranzke) [inline pkg/apis/garden/types.go:1205]: //+optional missing

File: pkg/apis/garden/types.go [hunk 4]
```diff
@@ -1200,11 +1200,12 @@ type Hibernation struct {
...
-	Start string
+	Start *string
 	// End is a Cron spec at which time a Shoot will be woken up.
```
Comment (rfranzke) [inline pkg/apis/garden/types.go:1208]: //+optional missing

File: pkg/apis/garden/v1beta1/types.go [hunk 5]
```diff
@@ -1192,11 +1192,12 @@ type Hibernation struct {
...
-	Start string `json:"start"`
+	Start *string `json:"start"`
 	// End is a Cron spec at which time a Shoot will be woken up.
```
Comment (rfranzke) [inline pkg/apis/garden/v1beta1/types.go:1200]: //+optional missing

File: pkg/apis/garden/v1beta1/types.go [hunk 6]
```diff
@@ -1192,11 +1192,12 @@ type Hibernation struct {
...
-	End string `json:"end"`
+	End *string `json:"end"`
```
Comment (rfranzke) [inline pkg/apis/garden/v1beta1/types.go]: omitempty

**Review by rfranzke — CHANGES_REQUESTED** [review]

**Comment by adracus** [comment]
Sorry, forgot the `omitempty` / `optional` - was already fixing it.

File: pkg/apis/garden/v1beta1/types.go (hunk 2 above)
Comment (rfranzke) [inline pkg/apis/garden/v1beta1/types.go:1197]: // +optional is still missing

File: pkg/apis/garden/v1beta1/types.go [hunk 7]
```diff
@@ -1192,11 +1192,12 @@ type Hibernation struct {
 
 // HibernationSchedule determines the hibernation schedule of a Shoot.
 // A Shoot will be regularly hibernated at each start time and will be woken up at each end time.
+// Start or End can be omitted, though at least one of each has to be specified.
 type HibernationSchedule struct {
 	// Start is a Cron spec at which time a Shoot will be hibernated.
-	Start string `json:"start"`
+	Start *string `json:"start,omitempty"`
 	// End is a Cron spec at which time a Shoot will be woken up.
```
Comment (rfranzke) [inline pkg/apis/garden/v1beta1/types.go:1200]: // +optional is still missing

**Review by rfranzke — CHANGES_REQUESTED** [review]

**Review by rfranzke — APPROVED** [review]
/lgtm

**Pull request merged**